backend/tracim_backend/views/core_api/workspace_controller.py

```diff
@@ -99,7 +99,11 @@ def workspaces(self, context, request: TracimRequest, hapic_data=None):
 
     @hapic.with_api_doc(tags=[SWAGGER_TAG_WORKSPACE_ENDPOINTS])
     @hapic.handle_exception(EmptyLabelNotAllowed, HTTPStatus.BAD_REQUEST)
-    @require_workspace_role(UserRoleInWorkspace.WORKSPACE_MANAGER)
+    @require_profile_or_other_profile_with_workspace_role(
```

Comment:
`require_profile_or_other_profile_with_workspace_role()`

What I understand is :

- if user is Admin then ok
- if user is User and WorkspaceManager then ok

So...

Why not writting a decorator : `require_workspace_role(UserRoleInWorkspace.WORKSPACE_MANAGER, allow_superuser=True)` ?

Sure its less flexible, but its immediate to understand

Reply:
Decorator used was created for one case where we need this flexibility : for workspace deletion case where only admin and trusted_user who are workspace_manager role can delete workspace. 

I can create a new decorator if you wish but require_profile_or_other_profile_with_workspace_role is needed.

Reply:
@inkhey 

you tell that a workspace can be deleted : 
- if the user is superadmin
- if the user is trusted-user + is the workspace manager

My feeling was a workspace can be deleted : 
- if the user is superadmin
- if the user is workspace manager.

Am I wrong? (its possible).

so maybe the decorator should be : `require_profile_and_role(profile=xxx, role=yyy, allow_superuser=False)`

Could you list here the require_XXX decorators?

Reply:
@lebouquetin it was previousiy decided that a workspace manager who have not the right to create workspace can't delete any workspace, because it's not at the same level than modifying thing inside workspace.

decorator list : 
```
# profile
require_same_user_or_profile(group: int)
require_profile(group: int)
require_profile_or_other_profile_with_workspace_role(allow_all_group: int, allow_if_role_group: 
int, minimal_required_role: int)
# role
require_workspace_role(minimal_required_role: int)
require_candidate_workspace_role(minimal_required_role: int) # check another field than workspace role
# content_type
require_content_types(content_types_slug: typing.List[str])
# specific for comment
require_comment_ownership_or_role(minimal_required_role_for_owner: int, minimal_required_role_for_anyone: int,)

```